library/src/contraction/contraction_selection.cpp

```diff
@@ -2602,7 +2595,18 @@ namespace hiptensor
 
             if(auto candidate = candidates.find(unique_id); candidate != candidates.end())
             {
-                *winner = candidate->second;
+                auto* solution = candidate->second;
+
+                PerfMetrics metrics = {
```

Comment:
Can probably remove the metrics as they are unused. Then can condense the assignment of winner to candidate->second